### ISSUE TITLE
Training

### ISSUE THREAD

**Issue by afsarsal** [NsfwSpy/NsfwSpy.NET#19]
Hi,

Thanks for sharing this, it's pretty useful.

Could you advise how I can add additional classification?
I would like to add and classify gambling related images.

Also I've tried to run the Train project but getting an error on following line;

var trainedModel = trainingPipeline.Fit(trainSet);

And the error message is;

System.ArgumentOutOfRangeException: 'Only one class found in the LabelAsKey column. To build a multiclass classification model, the number of classes needs to be 2 or greater Arg_ParamName_Name'

Many thanks

**Comment by d00ML0rDz** [comment]
Hey,

Have you made any modifications to the training code at all? On line 18 of the Program.cs file, there should be this line:
```
var assetsPath = @"E:\NsfwSpy\Images";
```
This path will need to be changed to the directory of the images folder you want to train with on your local machine which will contain children folders named after each label (Hentai, Neutral, Pornography, Sexy, Gambling) and containing images of that classification type. We currently don't publicly provide our training dataset, so you will need to source your own if you want to also classify all of the labels that NsfwSpy currently can.

The "Gambling" label will then need to be added alongside the other labels and then hopefully the training process will work 🙂 Feel free to comment here if you have any further questions or need me to clarify anything further.

**Comment by d00ML0rDz** [comment]
Is it possible for you to let me know what path you set on line 18 of the Program.cs file and screenshot the folders inside of that please?

**Comment by afsarsal** [comment]
https://github.com/NsfwSpy/NsfwSpy.NET/assets/30019998/bb1af5ad-50fc-479f-b7a5-e8d2d9da4713

Yes sure, please find attached video. Let me know if you need anything else. Thank you again

**Comment by d00ML0rDz** [comment]
Cheers for the video, makes it much clearer. Could you try with just .jpg or .png files in those folders? I don't think that ML.NET supports .webp images for training.

**Comment by afsarsal** [comment]
aah thanks, the issue was on following function which was skipping webp files. Now I've added webp format and it looks it's processing the files.
Many thanks again.

public static IEnumerable<ImageData> LoadImagesFromDirectory(string folder)
        {
            var allowedFileExtensions = new[] { ".jpg", ".jpeg", ".png", ".**webp**" };
            var files = Directory.GetFiles(folder, "*", searchOption: SearchOption.AllDirectories);

**Comment by d00ML0rDz** [comment]
Glad that's working for you. I haven't seen any information regarding whether ML.NET now supports .webp files when training. I think when classifying .webp files, NsfwSpy has to convert them into a different format beforehand. Just something to be mindful of 😊
